uber/site_sections/accounts.py

```diff
@@ -88,7 +88,7 @@ def reset(self, session, message='', email=None):
             try:
                 account = session.get_account_by_email(email)
             except NoResultFound:
-                message = 'No account exists for email address {!r}'.format(email)
+                message = 'The ID/Password combination provided is not valid {!r}'.format(email)
```

Comment:
I was actually thinking it would be better just to eliminate the
NoResultFound section, since we NEVER want to tell someone trying to break
in that this was the reason their attempt failed.

And then I didn't delete it.  Hmmm...

I don't see anywhere that we tell them the e-mail attempt was successful.
Am I missing something?

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

On Mon, Aug 3, 2015 at 4:43 PM, Eli Courtwright <notifications@github.com>
wrote:

> In uber/site_sections/accounts.py
> <https://github.com/magfest/ubersystem/pull/1342#discussion_r36129169>:
>
> > @@ -88,7 +88,7 @@ def reset(self, session, message='', email=None):
> >              try:
> >                  account = session.get_account_by_email(email)
> >              except NoResultFound:
> > -                message = 'No account exists for email address {!r}'.format(email)
> > +                message = 'The ID/Password combination provided is not valid {!r}'.format(email)
>
> This particular change is insufficient, because on the reset page they
> ONLY enter an email address. So if we want people to not know what email
> addresses exist, then instead of returning this message (which tells them
> conclusively that the email address does not exist), we should instead just
> do a
>
> raise HTTPRedirect('login?message={}', 'Your new password has been emailed to you')
>
> which is what we do down below if the reset is successful. If we give a
> different response for a valid vs an invalid email address then there's no
> point.
>
> If we want to make it slightly less deceptive, then we can change the
> message in both places to read something like "A reset password has been
> mailed to that address, if that address exists in our system." That way
> we're not outright fooling people who think they have accounts into
> believing they've been emailed a reset when in fact they have not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/magfest/ubersystem/pull/1342/files#r36129169>.
>